### ISSUE TITLE
Issues when XALT is loaded and Python from Lmod are loaded

### ISSUE THREAD

**Issue by treydock** [xalt/xalt#38]
We are running XALT 2.9.10 on RHEL 7.7 and have had users report issues like the following:

```
$ module load python/3.6-conda5.2
$ module list

Currently Loaded Modules:
  1) xalt/latest   2) gcc-compatibility/8.4.0   3) intel/19.0.5   4) mvapich2/2.3.3   5) modules/sp2020   6) python/3.6-conda5.2

 

$ python
Python 3.6.6 |Anaconda custom (64-bit)| (default, Jun 28 2018, 17:14:51) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import matplotlib.pyplot as plt
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/apps/python/3.6-conda5.2/lib/python3.6/site-packages/matplotlib/pyplot.py", line 31, in <module>
    import matplotlib.colorbar
  File "/apps/python/3.6-conda5.2/lib/python3.6/site-packages/matplotlib/colorbar.py", line 36, in <module>
    import matplotlib.contour as contour
  File "/apps/python/3.6-conda5.2/lib/python3.6/site-packages/matplotlib/contour.py", line 20, in <module>
    import matplotlib.font_manager as font_manager
  File "/apps/python/3.6-conda5.2/lib/python3.6/site-packages/matplotlib/font_manager.py", line 58, in <module>
    from matplotlib import afm, cbook, ft2font, rcParams, get_cachedir
ImportError: /apps/xalt/2.9.10/lib64/libz.so.1: version `ZLIB_1.2.9' not found (required by /apps/python/3.6-conda5.2/lib/python3.6/site-packages/matplotlib/../../.././libpng16.so.16)
>>> 

$ echo $LD_LIBRARY_PATH
/apps/python/3.6-conda5.2/lib:/opt/mvapich2/intel/19.0/2.3.3/lib:/apps/gnu/8.4.0/lib64:/apps/gnu/8.4.0/lib:/opt/intel/19.0.5/debugger_2019/libipt/intel64/lib:/opt/intel/19.0.5/compilers_and_libraries_2019/linux/lib/intel64_lin:/opt/intel/19.0.5/compilers_and_libraries_2019/linux/daal/lib/intel64_lin:/opt/intel/19.0.5/compilers_and_libraries_2019/linux/ipp/lib/intel64_lin:/opt/intel/19.0.5/compilers_and_libraries_2019/linux/mkl/lib/intel64_lin:/opt/intel/19.0.5/compilers_and_libraries_2019/linux/tbb/lib/intel64_lin/gcc4.4

$ ls /apps/python/3.6-conda5.2/lib/libz*
/apps/python/3.6-conda5.2/lib/libz.a     /apps/python/3.6-conda5.2/lib/libzmq.so        /apps/python/3.6-conda5.2/lib/libz.so
/apps/python/3.6-conda5.2/lib/libzmq.a   /apps/python/3.6-conda5.2/lib/libzmq.so.5      /apps/python/3.6-conda5.2/lib/libz.so.1
/apps/python/3.6-conda5.2/lib/libzmq.la  /apps/python/3.6-conda5.2/lib/libzmq.so.5.1.5  /apps/python/3.6-conda5.2/lib/libz.so.1.2.11
```

**Comment by rtmclay** [comment]
Please copy /apps/python/3.6-conda5.2/lib/libz.so.1.2.11 to /apps/xalt/2.9.10/lib64 to see if it fixes the problem.

**Comment by treydock** [comment]
That cluster using /apps has no test environment at this time and so I moved this reproduction step to a different cluster where `/apps` becomes `/usr/local` but XALT version and OS are the same versions.

Appears like the issue does not go away with copying libz.so.1.2.11.

```
$ cp /usr/local/python/3.6-conda5.2/lib/libz.so.1.2.11 /usr/local/xalt/2.9.10/lib64/
$ ls  /usr/local/xalt/2.9.10/lib64/*
/usr/local/xalt/2.9.10/lib64/libcom_err.so.2      /usr/local/xalt/2.9.10/lib64/libnssutil3.so
/usr/local/xalt/2.9.10/lib64/libcrypto.so         /usr/local/xalt/2.9.10/lib64/libpcre.so.1
/usr/local/xalt/2.9.10/lib64/libcrypto.so.0.9.8e  /usr/local/xalt/2.9.10/lib64/libplc4.so
/usr/local/xalt/2.9.10/lib64/libcrypto.so.10      /usr/local/xalt/2.9.10/lib64/libplds4.so
/usr/local/xalt/2.9.10/lib64/libcrypto.so.1.0.2k  /usr/local/xalt/2.9.10/lib64/libpthread.so.0
/usr/local/xalt/2.9.10/lib64/libcrypto.so.6       /usr/local/xalt/2.9.10/lib64/libresolv.so.2
/usr/local/xalt/2.9.10/lib64/libcrypt.so.1        /usr/local/xalt/2.9.10/lib64/librt.so.1
/usr/local/xalt/2.9.10/lib64/libcurl.so           /usr/local/xalt/2.9.10/lib64/librun_submission.a
/usr/local/xalt/2.9.10/lib64/libcurl.so.4         /usr/local/xalt/2.9.10/lib64/libsasl2.so.3
/usr/local/xalt/2.9.10/lib64/libcurl.so.4.3.0     /usr/local/xalt/2.9.10/lib64/libselinux.so.1
/usr/local/xalt/2.9.10/lib64/libfreebl3.so        /usr/local/xalt/2.9.10/lib64/libsmime3.so
/usr/local/xalt/2.9.10/lib64/libgcc_s.so.1        /usr/local/xalt/2.9.10/lib64/libssh2.so.1
/usr/local/xalt/2.9.10/lib64/libgssapi_krb5.so.2  /usr/local/xalt/2.9.10/lib64/libssl3.so
/usr/local/xalt/2.9.10/lib64/libidn.so.11         /usr/local/xalt/2.9.10/lib64/libssl.so.10
/usr/local/xalt/2.9.10/lib64/libk5crypto.so.3     /usr/local/xalt/2.9.10/lib64/libstdc++.so.6
/usr/local/xalt/2.9.10/lib64/libkeyutils.so.1     /usr/local/xalt/2.9.10/lib64/libuuid.so
/usr/local/xalt/2.9.10/lib64/libkrb5.so.3         /usr/local/xalt/2.9.10/lib64/libuuid.so.1
/usr/local/xalt/2.9.10/lib64/libkrb5support.so.0  /usr/local/xalt/2.9.10/lib64/libuuid.so.1.3.0
/usr/local/xalt/2.9.10/lib64/liblber-2.4.so.2     /usr/local/xalt/2.9.10/lib64/libxalt_init.so
/usr/local/xalt/2.9.10/lib64/libldap-2.4.so.2     /usr/local/xalt/2.9.10/lib64/libz.so.1
/usr/local/xalt/2.9.10/lib64/libm.so.6            /usr/local/xalt/2.9.10/lib64/libz.so.1.2.11
/usr/local/xalt/2.9.10/lib64/libnspr4.so          /usr/local/xalt/2.9.10/lib64/xalt_initialize_regular.o
/usr/local/xalt/2.9.10/lib64/libnss3.so

$ module load python/3.6-conda5.2
$ module list

Currently Loaded Modules:
  1) xalt/latest   2) gcc-compatibility/8.4.0   3) intel/19.0.5   4) mvapich2/2.3.3   5) modules/sp2020   6) python/3.6-conda5.2

$ python
Python 3.6.6 |Anaconda custom (64-bit)| (default, Jun 28 2018, 17:14:51) 
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import matplotlib.pyplot as plt
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/python/3.6-conda5.2/lib/python3.6/site-packages/matplotlib/pyplot.py", line 31, in <module>
    import matplotlib.colorbar
  File "/usr/local/python/3.6-conda5.2/lib/python3.6/site-packages/matplotlib/colorbar.py", line 36, in <module>
    import matplotlib.contour as contour
  File "/usr/local/python/3.6-conda5.2/lib/python3.6/site-packages/matplotlib/contour.py", line 20, in <module>
    import matplotlib.font_manager as font_manager
  File "/usr/local/python/3.6-conda5.2/lib/python3.6/site-packages/matplotlib/font_manager.py", line 55, in <module>
    from matplotlib import afm, cbook, ft2font, rcParams, get_cachedir
ImportError: /usr/local/xalt/2.9.10/lib64/libz.so.1: version `ZLIB_1.2.9' not found (required by /usr/local/python/3.6-conda5.2/lib/python3.6/site-packages/matplotlib/../../.././libpng16.so.16)
>>> 

```

These are the steps I take to build XALT:

```
module purge
./configure --prefix=/usr/local \
--with-etcDir=/usr/local/xalt/etc \
--with-transmission=syslog \
--with-trackGPU=yes \
--with-syshostConfig=hardcode:$LMOD_SYSTEM_NAME \
--with-config=Config/OSC_config.py \
--with-staticLibs=yes \
--with-MySQL=no
make install
```

**Comment by rtmclay** [comment]
XALT 2.9.10 is designed to support containers which might not have a zlib.  It does so by adding RPATH to the XALT library so it can find things in /usr/local/xalt/2.9.10/lib64.  I don't see another way to support containers when I use zlib.   I think that you will have to go back to XALT 2.9.7 to avoid this.   It looks like I'm going to have use a different compression library than zlib but that may take some time (say two weeks) for me to fix it.

**Comment by treydock** [comment]
Downgrading to XALT 2.9.7 solved this particular issue.  I wonder if zlib is just one of many that will trigger this library symbol issue?  I don't know if the issue is specific to just zlib or could manifest itself if things are linked to newer versions of shared libraries that are copied to XALT's lib64 directory.

**Comment by ZQyou** [comment]
With XALT 2.9.7, I got this error
```
singularity run shub://singularityhub/hello-world
INFO:    Use cached image
/bin/sh: error while loading shared libraries: libcurl.so.4: cannot open shared object file: No such file or directory

**Comment by treydock** [comment]
@rtmclay In case it's not clear, @ZQyou and I work at same center.  I downgraded our test environment to XALT 2.9.7 and now the errors are with singularity when XALT module is loaded per ZQ's comment.

**Comment by ZQyou** [comment]
Reposted the singularity issue with 2.9.7
```
$ singularity version
3.6.4-1.el7

$ singularity run shub://singularityhub/hello-world
INFO:    Use cached image
/bin/sh: error while loading shared libraries: libcurl.so.4: cannot open shared object file: No such file or directory
```

**Comment by rtmclay** [comment]
I can't solve both problems at the same time for the moment. Currently I am thinking that I'll have to find a separate library that compresses strings.  But that means that I'll have to get it to work as a python extension as well.

**Comment by rtmclay** [comment]
Having time to think about it, I found a way to remove the dependency on libz.  I have also removed the RPATH to XALT's lib64 directory. Please test XALT version 2.9.11. This version should also fix the problem with containers and with python failing to find the libz it wanted.

**Comment by treydock** [comment]
I deployed XALT 2.9.11 and that appears to have resolved the issues as well a new issue that came in with getting the wrong libcurl.so while we were on XALT 2.9.7.

**Comment by ZQyou** [comment]
It works fine so far. We can close this issue now. Thank you.

**Comment by rtmclay** [comment]
I have just released XALT 2.10.  I just found a problem where the XALT version 2.9+ the XALT watermark was ignored even when it existed. This means that if you care about what compiler was used for executable that information was lost.  Since I know that you are running XALT 2.9.11, I thought I'd let you know directly.
